### PR TITLE
Fix unpermitted param  during checkout payment update to fix #6203

### LINKED ISSUE #6203: Unpermitted parameter: existing_card during checkout
When running through the checkout process in development with `config.action_controller.action_on_unpermitted_parameters = :raise` after choosing to checkout with an existing card (using Spree::Gateway::Bogus) the following error is thrown:
```
ActionController::UnpermittedParameters at /checkout/update/payment
found unpermitted parameter: existing_card
```

This is the result of `existing_card` being sent in the params and later removed (used only to retrieve/set the payment method). However when `.permit` is run on [line 232](https://github.com/spree/spree/blob/master/core/app/models/spree/order/checkout.rb#L232) of the Spree::Order Checkout module `:existing_card` is not whitelisted so it logs or throws an error depending on your settings.

### PR DESCRIPTION
**Pull request by farneman**
Retrieve the `:existing_card` param and remove it from the params hash before calling `.permit`.

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage to preview changes. <a href='http://farneman-prevent-unpermitted-existing-card-param.spree-spree-9ed724b77adec04c49c2.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://farneman-prevent-unpermitted-existing-card-param.spree-spree-9ed724b77adec04c49c2.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

**Comment by gmacdougall** [comment]
Thanks for your effort on this. Could you please create a spec to show that this issue was fixed in order to prevent a potential regression?

**Comment by farneman** [comment]
Sure thing.

**Comment by farneman** [comment]
The only way I can see to be able to accurately test it is to set `  config.action_controller.action_on_unpermitted_parameters = :raise` in the test.rb for core's dummy app. This results in 9 other test failures (most from checkout.rb related tests) that would need to be reviewed an fixed.

**Comment by JDutil** [comment]
You should be able to just update the configuration in a before block, and then undo it in an after block.  It would probably be worthwhile to deal with the other 9 failures though as you're probably just going to run into those as issues as well.

**Comment by farneman** [comment]
Thanks @JDutil I've added a test for this specific issue with the config change in before/after blocks. I'll look into the other param related issues and make a separate PR(s) for those as necessary.

File: core/app/models/spree/order/checkout.rb [hunk 1]
```diff
@@ -229,11 +229,13 @@ def update_from_params(params, permitted_params, request_env = {})
...
             @updating_params = params
             run_callbacks :updating_from_params do
+              existing_card_id = @updating_params[:order] ? @updating_params[:order].delete(:existing_card) : nil
+
               attributes = @updating_params[:order] ? @updating_params[:order].permit(permitted_params).delete_if { |k,v| v.nil? } : {}
 
               # Set existing card after setting permitted parameters because
               # rails would slice parameters containg ruby objects, apparently
-              existing_card_id = @updating_params[:order] ? @updating_params[:order][:existing_card] : nil
+             #  existing_card_id = @updating_params[:order] ? @updating_params[:order][:existing_card] : nil
```
Comment (JDutil) [inline core/app/models/spree/order/checkout.rb]: Could you please remove this line altogether now and move the comments above where you've moved this line to.

Comment (JDutil) [inline core/app/models/spree/order/checkout.rb]: I got it nevermind

**Comment by JDutil** [comment]
Merged w/ 6e018f1114a7ea04deae2e7b68971e9eaa6c5bf9

**Pull request closed without merging**